components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIMWSDLReader.java

```diff
@@ -139,10 +151,93 @@ public OMElement readAndCleanWsdl2(API api) throws APIManagementException {
+     */
+    public void validateBaseURI() throws APIManagementException {
+        if (baseURI.startsWith(APIConstants.WSDL_REGISTRY_LOCATION_PREFIX)) {
+            baseURI = APIUtil.getServerURL() + baseURI;
+        }
+
+        boolean isWsdl20 = false;
+        boolean isWsdl11 = false;
+
+        BufferedReader in = null;
+        try {
+            String inputLine;
+            StringBuilder urlContent = new StringBuilder();
+            URL wsdl = new URL(baseURI);
+            in = new BufferedReader(new InputStreamReader(wsdl.openStream(), Charset.defaultCharset()));
+            while ((inputLine = in.readLine()) != null) {
+                urlContent.append(inputLine);
+                isWsdl20 = urlContent.indexOf(WSDL20_NAMESPACE) > 0;
+                isWsdl11 = urlContent.indexOf(WSDL11_NAMESPACE) > 0;
+            }
+        } catch (IOException e) {
+            throw new APIManagementException("Error while reading WSDL from base URI " + baseURI, e);
+        } finally {
+            IOUtils.closeQuietly(in);
+        }
+
+        try {
+            if (isWsdl11) {
+                readAndValidateWSDL11();
+            } else if (isWsdl20) {
+                readAndValidateWSDL20();
+            } else {
+                throw new APIManagementException("URL is not in format of wsdl1.1 or wsdl2.0");
+            }
+        } catch (WSDLException e) {
+            throw new APIManagementException("Error while parsing WSDL content", e);
+        } catch (org.apache.woden.WSDLException e) {
+            throw new APIManagementException("Error while parsing WSDL content", e);
+        }
+    }
+
+    /**
+     * Given a URL, this method checks if the underlying document is a WSDL2
+     *
+     * @return true if the underlying document is a WSDL2
+     * @throws APIManagementException if error occurred while checking whether baseURI is WSDL2.0
+     */
+    public boolean isWSDL2BaseURI() throws APIManagementException {
+        URL wsdl;
+        boolean isWsdl2 = false;
+        BufferedReader in = null;
+        try {
+            wsdl = new URL(baseURI);
+            in = new BufferedReader(new InputStreamReader(wsdl.openStream(), Charset.defaultCharset()));
+
+            String inputLine;
+            StringBuilder urlContent = new StringBuilder();
+            while ((inputLine = in.readLine()) != null) {
+                urlContent.append(inputLine);
+                isWsdl2 = urlContent.indexOf(WSDL20_NAMESPACE) > 0;
+            }
+        } catch (MalformedURLException e) {
+            throw new APIManagementException("Malformed URL encountered", e);
+        } catch (IOException e) {
+            throw new APIManagementException("Error Reading Input from Stream from " + baseURI, e);
+        } finally {
+            IOUtils.closeQuietly(in);
+        }
+
+        try {
+            if (isWsdl2) {
+                readAndValidateWSDL20();
+            }

```

Comment:
shall we add a comment explaining this logic? i.e. if 1.2 why 2.0?

Reply:
That is actually Wsd"L"2 :)